### PR TITLE
fix 'catastrophic backtracking' issue with text-input regex

### PR DESCRIPTION
**Pull request by SanjalKatiyar**
https://bugzilla.redhat.com/show_bug.cgi?id=2270446

### CODE REVIEW

File: packages/shared/src/utils/validation.ts [hunk 1]
```diff
@@ -1,14 +1,13 @@
 /**
  * Starts and ends with a lowercase letter or number
  */
-export const startAndEndsWithAlphanumerics =
-  /^(?![\W])([a-z0-9]|[\W]*([a-z0-9]))+(?![\W])$/;
```
Comment (SanjalKatiyar) [inline packages/shared/src/utils/validation.ts:5]: this was causing `catastrophic backtracking` for certain format of input strings...

File: packages/shared/src/utils/validation.ts [hunk 2]
```diff
@@ -1,14 +1,13 @@
...
+export const startAndEndsWithAlphanumerics = /^[a-z0-9](.*[a-z0-9])?$/;
 
 /**
  * Only lowercase letters, numbers, non-consecutive periods, or hyphens
  */
 export const alphaNumericsPeriodsHyphensNonConsecutive =
-  /^[a-z0-9]+([a-z0-9]|([-.](?![-.])))*[a-z0-9]*$/;
```
Comment (SanjalKatiyar) [inline packages/shared/src/utils/validation.ts:11]: this is just minor enhancement, earlier it was returning false for strings prefixed with `-` or `.`, whereas it should pass for such cases and satisfy `Only lowercase letters, numbers, non-consecutive periods, or hyphens`.

**Review by alfonsomthd — APPROVED** [review]
It would be great to add some unit tests to ensure the behavior.

**Comment by SanjalKatiyar** [comment]
> It would be great to add some unit tests to ensure the behavior.

ack, will add as part of some other PR.

**Pull request merged**